docs/manual/src/internals/bindings_render_pipeline.md

```diff
@@ -0,0 +1,307 @@
+# Rendering Foreign Bindings
+
+UniFFI models foreign bindings generation like a compiler pipeline:
+
+```mermaid
+flowchart LR
+    ProcMacros[Rust + proc-macros] --> Metadata[Metadata]
+    UDL[UDL] --> Metadata[Metadata]
+    Metadata --> BindingsIr[Bindings IR]
+    BindingsIr --> KotlinIr[Kotlin IR]
+    BindingsIr --> SwiftIr[Swift IR]
+    BindingsIr --> PythonIr[Python IR]
+    BindingsIr --> RubyIr[Ruby IR]
+    KotlinIr --> Kotlin[Generated Kotlin]
+    SwiftIr --> Swift[Generated Swift]
+    PythonIr --> Python[Generated Python]
+    RubyIr --> Ruby[Generated Ruby]
+```
+
+- First, metadata is generated from the proc-macro code and UDL files.
+  The metadata is a simple reflection of the Rust definitions.
+- Then the metadata is transformed into an intermediate representation called the Bindings IR.
+- Then, for each bindings language:
+    - The Bindings IR is specialized for that language.
+    - The language-specific IR is used to generate the bindings code.
+
+## Proc-macro Metadata Generation
+
+When a UniFFI proc-macro wraps a Rust item, it records metadata about the item it's wrapping.
+For example, when `#[uniffi::export]` wraps a function, it records the function name, arguments, return type, etc.
+This metadata is then serialized into a buffer and stored as an exported symbol in the Rust library.
+The generated code will look something like this:
+
+```rust
+// Medata buffer for the `arithmetic::add()` function
+const UNIFFI_META_CONST_ARITHMETIC_FUNC_ADD: ::uniffi::MetadataBuffer = ::uniffi::MetadataBuffer::from_code(
```

Comment:
nit: it might be helpful to also show the add function that generated this metadata